android/demos/compose/src/main/kotlin/com/uber/rib/compose/root/main/logged_in/tic_tac_toe/TicTacToeInteractor.kt

```diff
@@ -36,58 +37,55 @@ class TicTacToeInteractor(
+            val coord = it.coordinate
 
-        if (board.cells[coord.x][coord.y] == null) {
-          if (currentPlayer == Board.MarkerType.CROSS) {
-            board.cells[coord.x][coord.y] = Board.MarkerType.CROSS
-            board.currentRow = coord.x
-            board.currentCol = coord.y
-            currentPlayer = Board.MarkerType.NOUGHT
-          } else {
-            board.cells[coord.x][coord.y] = Board.MarkerType.NOUGHT
-            board.currentRow = coord.x
-            board.currentCol = coord.y
-            currentPlayer = Board.MarkerType.CROSS
-          }
-        }
+            if (board.cells[coord.x][coord.y] == null) {
+              if (currentPlayer == Board.MarkerType.CROSS) {
+                board.cells[coord.x][coord.y] = Board.MarkerType.CROSS
+                board.currentRow = coord.x
+                board.currentCol = coord.y
+                currentPlayer = Board.MarkerType.NOUGHT
+              } else {
+                board.cells[coord.x][coord.y] = Board.MarkerType.NOUGHT
+                board.currentRow = coord.x
+                board.currentCol = coord.y
+                currentPlayer = Board.MarkerType.CROSS
+              }
+            }
 
-        if (board.hasWon(Board.MarkerType.CROSS)) {
-          listener.onGameWon(authInfo.playerOne)
-        } else if (board.hasWon(Board.MarkerType.NOUGHT)) {
-          listener.onGameWon(authInfo.playerTwo)
-        } else if (board.isDraw()) {
-          listener.onGameWon(null)
-        }
+            if (board.hasWon(Board.MarkerType.CROSS)) {
+              listener.onGameWon(authInfo.playerOne)
+            } else if (board.hasWon(Board.MarkerType.NOUGHT)) {
+              listener.onGameWon(authInfo.playerTwo)
+            } else if (board.isDraw()) {
+              listener.onGameWon(null)
+            }
 
-        val newPlayerName = if (currentPlayer == Board.MarkerType.CROSS) {
-          authInfo.playerOne
-        } else {
-          authInfo.playerTwo
-        }
+            val newPlayerName = if (currentPlayer == Board.MarkerType.CROSS) {
+              authInfo.playerOne
+            } else {
+              authInfo.playerTwo
+            }
 
-        stateStream.dispatch(
-          stateStream.current().copy(
-            board = board,
-            currentPlayer = newPlayerName
-          )
-        )
-      }
-
-    eventStream.observe()
-      .ofType<TicTacToeEvent.XpButtonClick>()
-      .autoDispose(this)
-      .subscribe {
-        // router.goToSomewhere()
-      }
+            stateStream.dispatch(
+              stateStream.current().copy(
+                board = board,
+                currentPlayer = newPlayerName
+              )
+            )
+          }
+          TicTacToeEvent.XpButtonClick -> TODO("Go somewhere")
+        }
+      }.launchIn(coroutineScope)
   }
 
   interface Listener {
-    fun onGameWon(winnerName: String?)
+    suspend fun onGameWon(winnerName: String?)
```

Comment:
why does this need to be suspend?

Reply:
This was a holdover from a previous experiment with the interactor baseclass. We can remove this now as long as we launch a coroutine in the impl to emit to the sharedflow.